### PR TITLE
fix: protect Raw Logs queries from excessively large line limits

### LINKED ISSUE #613: VL-datasource: protect Raw Logs queries from excessively large line limits
### Describe the bug

In the VictoriaLogs Grafana datasource, `Raw Logs` queries allow setting arbitrarily large `Line limit` / `maxLines` values.

`vmui` protects the interactive UI with a hard upper bound (10,000 log lines) and a warning flow.

This can lead to:
- Browser tab freezes or crashes when too many log rows are rendered
- Backend memory spikes or, in the worst case, backend crashes

### Expected behavior

The datasource should protect the interactive UI from unsafe line limits. E.g., 

- Use a hard upper bound for `Raw Logs` line limit, for example `10000`, same as `vmui`
- Show a warning/confirmation dialog before running very large log queries, same as `vmui`

### To Reproduce

1. Open Grafana Explore with `VL-datasource`
2. Select `Raw Logs`
3. Set a very large `Line limit` such as `1000000` or higher
4. Run a query that matches many logs

### Versions of VictoriaLogs datasource and VictoriaLogs backend

master

### Additional information

This is similar in spirit to the VictoriaLogs `vmui` issue:
- https://github.com/VictoriaMetrics/VictoriaLogs/issues/500

### PR DESCRIPTION
**Pull request by arturminchukov**
Related issue: #613 

### Describe Your Changes

Protect Raw Logs queries from excessively large line limits
A hard upper bound of 10000 lines is now enforced on both the per-query `Line limit` and the datasource-wide `Maximum lines` settings; a confirmation dialog warns before committing a value above 1000 lines.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres to [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/victoriametrics/contributing/#pull-request-checklist).
- [x] My change adheres to [VictoriaMetrics development goals](https://docs.victoriametrics.com/victoriametrics/goals/).

---
## Summary by cubic
Enforces a hard cap of 10,000 lines for Raw Logs queries and warns for large limits to prevent slowdowns. Applies to both per-query `Line limit` and datasource `Maximum lines`, addressing #613.

- **Bug Fixes**
  - Clamp `Line limit` and datasource `Maximum lines` to 10,000 in the UI and datasource.
  - Show a confirmation dialog for values over 1,000 (with “Don’t show again”); run only after confirm.
  - Disable “Run query” over the cap and show a tooltip; display an input error when invalid.
  - Clear the setting when a negative value is entered.
  - Add tests for the warning hook, query settings validation (including negative input), and datasource clamping.

<sup>Written for commit 078da6e61e90ae9e3bf082cb8d3678ee960f1dd8. Summary will update on new commits.</sup>

### CODE REVIEW

File: src/components/QueryEditor/QueryEditorOptions.tsx [hunk 1]
```diff
@@ -54,6 +56,29 @@ export const QueryEditorOptions = React.memo<Props>(({ app, query, maxLines, onC
...
+    }
+    onChange({ ...query, maxLines: newMaxLines });
+  };
+
+  const onMaxLinesBlur = () => {
+    if (query.maxLines !== undefined && query.maxLines > LOGS_LIMIT_WARNING_THRESHOLD) {
+      requestConfirmation(query.maxLines);
+    }
+
+    if (query.maxLines === undefined) {
```
[the remark this answers is not in the document]
Comment (arturminchukov) [reply src/components/QueryEditor/QueryEditorOptions.tsx:76]: function `requestConfirmation` runs the query depending on the validity of the value.

**Review by dmitryk-dk — APPROVED** [review]
